IMPLEMENTATION.adoc

```diff
@@ -0,0 +1,101 @@
+
+- link:https://github.com/kiali/kiali-operator/tree/v1.69.0/playbooks/kiali-deploy.yml[kiali-deploy.yml] - this is invoked when a new Kiali CR is created or an existing one is modified. This playbook determines which version is to be installed/updated and runs that version's corresponding `kiali-deploy` link:https://github.com/kiali/kiali-operator/tree/v1.69.0/roles[Ansible role]. If Kiali is to be upgraded (that is, if the Kiali CR's `spec.version` has been changed), this playbook will first invoke the `kiali-remove` Ansible role of the version specified in the `spec.version` previously declared in the Kiali CR (as found in the Kiali CR's `status` field); once the old Kiali is removed, the new version is installed by the execution of the `kiali-deploy` Ansible role of the new version to be installed (as found in the Kiali CR `spec.version`). Therefore, an "upgrade" is really just an "uninstall" followed by an "install".
+- link:https://github.com/kiali/kiali-operator/tree/v1.69.0/playbooks/kiali-remove.yml[kiali-remove.yml] - this is invoked when a Kiali CR has been removed. This playbook determines which version of Kiali is being uninstalled (as found in the now-deleted Kiali CR's `spec.version` field) and runs that version's corresponding `kiali-remove` link:https://github.com/kiali/kiali-operator/tree/v1.69.0/roles[Ansible role].
+- link:https://github.com/kiali/kiali-operator/tree/v1.69.0/playbooks/kiali-new-namespace-detected.yml[kiali-new-namespace-detected.yml] - this is invoked when a new namespace is created in the cluster. This playbook is simple and small. It's only job is to simply "touch" any and all existing Kiali CRs such that they will be immediately reconciled by the KO. ("Touching" in this context means adding/modifying an link:https://github.com/kiali/kiali-operator/blob/v1.69.0/playbooks/kiali-new-namespace-detected.yml#L30-L31[annotation to the Kiali CR] such that that modification will cause the KO to trigger a reconcilation). The playbook will only touch those Kiali CRs link:https://github.com/kiali/kiali-operator/blob/v1.69.0/playbooks/kiali-new-namespace-detected.yml#L34[if the namespace was created after the Kiali CR was created] and if the link:https://github.com/kiali/kiali-operator/blob/v1.69.0/playbooks/kiali-new-namespace-detected.yml#L21[Kiali CR was not modified or touched within the current minute]. This playbook enables a useful feature for those Kiali installations that are not given cluster-wide access but are given access to a set of namespaces defined by regular expressions (see `spec.deployment.accessible_namespaces`) or by Istio discovery selectors. In that case, when the KO reconciles the touched Kiali CRs, it will create the necessary Role/RoleBinding resources to give the Kiali installation access to the new namespace that was detected.
```

Comment:
```suggestion
- link:https://github.com/kiali/kiali-operator/tree/v1.69.0/playbooks/kiali-new-namespace-detected.yml[kiali-new-namespace-detected.yml] - this is invoked when a new namespace is created in the cluster. This playbook is simple and small. It's only job is to simply "touch" any and all existing Kiali CRs ("touching" in this context means adding/modifying an link:https://github.com/kiali/kiali-operator/blob/v1.69.0/playbooks/kiali-new-namespace-detected.yml#L30-L31[annotation to the Kiali CR] such that the modification will cause the KO to trigger a reconcilation). The playbook will only touch those Kiali CRs link:https://github.com/kiali/kiali-operator/blob/v1.69.0/playbooks/kiali-new-namespace-detected.yml#L34[if the namespace was created after the Kiali CR was created] and if the link:https://github.com/kiali/kiali-operator/blob/v1.69.0/playbooks/kiali-new-namespace-detected.yml#L21[Kiali CR was not modified or touched within the current minute]. This playbook enables a useful feature for those Kiali installations that are not given cluster-wide access but are given access to a set of namespaces defined by regular expressions (see `spec.deployment.accessible_namespaces`) or by Istio Discovery Selectors. In that case, when the KO reconciles the touched Kiali CRs, it will create the necessary Role/RoleBinding resources to give the Kiali installation access to the newly detected namespace.
```